utils.go

```diff
@@ -69,3 +70,59 @@ func MakeGremlinQuery(gremlinQuery GremlinQuery, argRegexP *regexp.Regexp) (stri
 	}
 	return fmt.Sprintf(gremlinQuery.Query, args...), nil
 }
+
+func InterfacesMatch(interface1, interface2 interface{}) bool {
+	switch interface1.(type) {
+	case map[string]interface{}:
+		val1Map := interface1.(map[string]interface{})
+		val2Map := interface2.(map[string]interface{})
+		if len(val1Map) != len(val2Map) {
+			return false
+		}
+		for k, val1 := range val1Map {
+			val2, ok := val2Map[k]
+			if !ok {
+				return false
+			}
+			if !InterfacesMatch(val1, val2) {
+				return false
+			}
+		}
+
+	case []interface{}:
+		val1List := interface1.([]interface{})
+		val2List := interface2.([]interface{})
```

Comment:
Is the valList type assertion necessary for interface1? Don't we need to check if the type assertion is ok?

Reply:
I checked and it is necessary for 1 (I was surprised too). Opened a new PR and checked type assertion for val2

Reply:
new PR looks good. FYI if you wanted to avoid the assertion for 1 you could _assign_ to a new var during the `switch` itself
```
	switch OOOH := i.(type) {
		...
	}
```

but i kinda like this better